cmd/execute.cpp

```diff
@@ -92,7 +95,10 @@ int main(int argc, char* argv[]) {
 
     SILKWORM_LOG(LogInfo) << "Starting block execution. DB: " << db_file << std::endl;
 
-    lmdb::DatabaseConfig db_config{db_path, *map_size};
+    lmdb::DatabaseConfig db_config{db_path};
+    if (map_size) {
```

Comment:
This imho is not correct.
if(map_size) tells if map_size has value or not.
What you might want is whether the value is zero or not (which is *map_size)

Reply:
My intention is to leave the default map_size in lmdb::DatabaseConfig as is (1TB by default) if no --lmdb.mapSize option is provided.

Reply:
If, on the other hand, the user explicitly passes "--lmbd.mapSize 0", that should be honoured and not overwritten.

Reply:
Though the user has no idea that 0 means something special w/o looking into the code. Perhaps we can support a special input token, e.g. "--lmbd.mapSize dynamic", instead of arguably abusing 0.

Reply:
> If, on the other hand, the user explicitly passes "--lmbd.mapSize 0", that should be honoured and not overwritten.

Zero mapsize won't be honored in anyway.

- If the datafile is new minimum allocation size is size of metapages for 2 system tables
- If the datafile already exists any mapSize **below** data-file-size can't be honored or you end up with a mapped size shorter than data with meta pages pointing to un-mapped region (and on Windows truncates the file)
- If the data file already exists any mapSize **above** data-file-size behaves differently on *nix vs windows: in first case has no effect on data-file size, on the latter it widens the file.

An lmdb.mapSize == 0 means : "Open the file with a map_size exactly matching data-file-size" which is actually what chaindb.cpp does for any value < data-file-size

Reply:
I understand that dealing with defaults is a PIA especially if we want to portability.

Correct me if I'm wrong : you see the omission of --lmdb.mapSize as "assume the default size hardcoded". This has a drawback on windows where we'll end up with min 1TB file sizes.

I see the omission of --lmdb.mapSize as "Open the datafile with a map-size == data-file-size" which produces a consistent result amonf Win and *nix i.e. the file will likely report MDB_MAPFULL if try to insert data when no space left.

Reply:
As an alternative we can set a directive conditional on chaindb.cpp to set the default to 0 for win and to 1TB for *nix

Reply:
I reiterate.

On Windows always `map_size == data-file-size`. You cannot set a map_size greater than available storage space on the volume.

On *nix you can set any map_size (eventually greater than disk or volume) and data-file-size will grow accordingly to effective data put in.

Reply:
Yes, that's right. IMHO we should have sensible defaults. 1TB is not perfect, but for the time being it works well for both Windows and Linux/macOS.
Another possibility is to make the dynamic map size to work better as the default. For instance, what of we set the map_size to 120%, not 100% the file size to avoid MDB_MAPFULL. Perhaps we can take some inspiration from MDBX. But that'll require some work.
All in all, I'd set the default to 1TB until the dynamic one is more robust if we want to invest time in the dynamic mode.

Reply:
OK, let's set the default to dynamic on Windows and to 1TB on Linux/macOS.